### ISSUE TITLE
Second granularity config does not compile

### ISSUE THREAD

**Issue by weixiyen** [c-rack/quantum-elixir#156]
It seems Crontab.CronExpression cannot be found

**Comment by c-rack** [comment]
@weixiyen could you please provide more information about this issue? Crontab.CronExpression is used in our unit tests, so if you clone this project and run `mix test`, what is the output?

**Comment by weixiyen** [comment]
Repro steps

1. Add Quantum to mix.exs
2. Add the second granularity example to config.exs (below)

```elixir
use Mix.Config

config :quantum, :your_app,
  cron: [
    news_letter: [
      schedule: %Crontab.CronExpression{extended: true, second: [5]}, # Runs every minute at second 5
      task: "MyApp.NewsLetter.send", # {MyApp.NewsLetter, :send} is supported too
      args: [:whatever]
    ]
  ]
```

Output: Crontab.CronExpression cannot be found.

Tests pass but not the example where I try to set second granularity from config.  Does it work for you outside of tests?

**Comment by ali-l** [comment]
An alternative approach is to add the job when the application starts:
```elixir
job = %Quantum.Job{schedule: %Crontab.CronExpression{extended: true, second: [5]}, task: "MyApp.NewsLetter.send", args: [:whatever]}
Quantum.add_job(:news_letter, job)
```

**Comment by maennchen** [comment]
Hmm... I didn't consider that mix can't load a struct when writing this code.

Is there a way to import code in the Config or is there no way to get this to work?

**Comment by c-rack** [comment]
Maybe we can change it to use just a plain map instead of Crontab.CronExpression, and convert that map to a Crontab.CronExpression at a later stage?

**Comment by maennchen** [comment]
I'm a little bit frustrated about the fact, that no custom sigils / structs can be used in a config... That has to work :S 

If there is really no way, i propose to switch back to the string and for the extended ones we use a tuple.

**Comment by c-rack** [comment]
You could post this question to elixir-dev mailing list to ask if there are any other options.

**Comment by maennchen** [comment]
@c-rack It doesn't seem like there'll be a solution to this soon. I propose to allow a special string like "e@ * *" for extended schedules in the configuration.

**Comment by maennchen** [comment]
@c-rack I don't like to use a plain map instead of the sigil because the struct is not meant to write by hand. It is also not fault tolerant. You will get no matching clause found errors inside the crontab code if you make a mistake.

**Comment by c-rack** [comment]
Have you asked elixir-dev mailing list? If yes, could you please link to that thread?

**Comment by maennchen** [comment]
@c-rack I talked to a few people on the Elixir Slack. There seems to be no way. I haven't written to the mailing list.

**Comment by maennchen** [comment]
Mailing-List Thread: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/elixir-lang-core/V5P3EJaRBBo/qpTJKTuxDgAJ

Feel free to comment there if I didn't represent the issue correctly.

**Comment by maennchen** [comment]
@c-rack I don't think my post will gain any traction soon.

**Comment by c-rack** [comment]
But it was worth a try 👍

**Comment by maennchen** [comment]
This will be solved (not beautiful, but it works) with the #174 solution. I will close this issue therefore.
